refactor(language_server): remove `OnceCell` from `WorkspaceWorker.root_uri`

Sysix opened the pull request:
The root_uri of a `WorkspaceWorker` will never change. I copied the `OnceCell` approach from the `main.rs`.
Now, I do not think this is really needed because the workers are `Mutex` locked behind `Backend.workspace_workers`.

Sysix commented:
* **#10898** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10898?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10898?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10898).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **May 8, 10:47 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(language_server): remove `OnceCell` from `WorkspaceWorker.root_uri` (#10898)

The root_uri of a `WorkspaceWorker` will never change. I copied the `OnceCell` approach from the `main.rs`.
Now, I do not think this is really needed because the workers are `Mutex` locked behind `Backend.workspace_workers`.
